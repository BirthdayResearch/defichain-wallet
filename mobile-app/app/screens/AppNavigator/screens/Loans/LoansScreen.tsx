import { useCallback, useEffect, useLayoutEffect, useState } from 'react'
import { TouchableOpacity, View } from 'react-native'
import { tailwind } from '@tailwind'
import { ThemedIcon, ThemedView } from '@components/themed'
import { Tabs } from '@components/Tabs'
import { Vaults } from './components/Vaults'
import { EmptyVault } from './components/EmptyVault'
import { SkeletonLoader, SkeletonLoaderScreen } from '@components/SkeletonLoader'
import { useDispatch, useSelector } from 'react-redux'
import { RootState } from '@store'
import { fetchLoanSchemes, fetchLoanTokens, fetchVaults, loanTokensSelector } from '@store/loans'
import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
import { useWalletContext } from '@shared-contexts/WalletContext'
import { LoanCards } from './components/LoanCards'
import { StackScreenProps } from '@react-navigation/stack'
import { LoanParamList } from './LoansNavigator'
import { HeaderSearchIcon } from '@components/HeaderSearchIcon'
import { HeaderSearchInput } from '@components/HeaderSearchInput'
import { debounce } from 'lodash'
import { LoanToken } from '@defichain/whale-api-client/dist/api/loan'

enum TabKey {
  BrowseLoans = 'BROWSE_LOANS',
  YourVaults = 'YOUR_VAULTS'
}

type Props = StackScreenProps<LoanParamList, 'LoansScreen'>

export function LoansScreen ({ navigation }: Props): JSX.Element {
  const { address } = useWalletContext()
  const blockCount = useSelector((state: RootState) => state.block.count)
  const {
    vaults,
    hasFetchedVaultsData,
    hasFetchedLoansData
  } = useSelector((state: RootState) => state.loans)
  const loans = useSelector((state: RootState) => loanTokensSelector(state.loans))
  const [activeTab, setActiveTab] = useState<string>(TabKey.YourVaults)
  const dispatch = useDispatch()
  const client = useWhaleApiClient()
  const onPress = (tabId: string): void => {
    if (tabId === TabKey.YourVaults) {
      setShowSearchInput(false)
    } else if (searchString !== '') {
      setShowSearchInput(true)
    } else {
      // no-op: maintain search input state if no query
    }
    setActiveTab(tabId)
  }
  const tabsList = [{
    id: TabKey.YourVaults,
    label: 'Your vaults',
    disabled: false,
    handleOnPress: onPress
  }, {
    id: TabKey.BrowseLoans,
    label: 'Browse loan tokens',
    disabled: false,
    handleOnPress: onPress
  }]

  // Search
  const [filteredLoans, setFilteredLoans] = useState<LoanToken[]>(loans)
  const [showSeachInput, setShowSearchInput] = useState(false)
  const [searchString, setSearchString] = useState('')
  const handleFilter = useCallback(
    debounce((searchString: string) => {
      setFilteredLoans(loans.filter(loan =>
        loan.token.displaySymbol.toLowerCase().includes(searchString.trim().toLowerCase())
      ))
    }, 500)
    , [loans, hasFetchedLoansData])

  useEffect(() => {
    if (loans.length === 0) {
      return
    }
    handleFilter(searchString)
  }, [searchString])

  useEffect(() => {
    setFilteredLoans(loans)
  }, [hasFetchedLoansData])

  useEffect(() => {
    dispatch(fetchVaults({
      address,
      client
    }))
    dispatch(fetchLoanTokens({ client }))
  }, [blockCount, address])

  useEffect(() => {
    dispatch(fetchLoanSchemes({ client }))
  }, [])

  useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: (): JSX.Element => {
        if (activeTab === TabKey.BrowseLoans && vaults.length !== 0) {
          return (
            <HeaderSearchIcon testID='header_loans_search' onPress={() => setShowSearchInput(true)} />
          )
        } else {
          return (
            <TouchableOpacity
              // eslint-disable-next-line
              onPress={() => navigation.navigate({
                name: 'CreateVaultScreen',
                params: {},
                merge: true
              })}
              testID='create_vault_header_button'
            >
              <ThemedIcon
                size={28}
                style={tailwind('mr-2')} light={tailwind('text-primary-500')}
                dark={tailwind('text-primary-500')} iconType='MaterialCommunityIcons' name='plus'
              />
            </TouchableOpacity>
          )
        }
      }
    })
  }, [navigation, activeTab, vaults])

  useEffect(() => {
    if (showSeachInput) {
      navigation.setOptions({
        header: (): JSX.Element => (
          <HeaderSearchInput
            searchString={searchString}
            onClearInput={() => setSearchString('')}
            onChangeInput={(text: string) => setSearchString(text)}
            onCancelPress={() => {
              setSearchString('')
              setShowSearchInput(false)
            }}
            placeholder='Search for loans'
            testID='loans_search_input'
          />
        )
      })
    } else {
      navigation.setOptions({
        header: undefined
      })
    }
  }, [showSeachInput, searchString])

  if (!hasFetchedVaultsData) {
    return (
      <View style={tailwind('mt-1')}>
        <SkeletonLoader
          row={3}
          screen={SkeletonLoaderScreen.Vault}
        />
      </View>
    )
  } else if (vaults?.length === 0) {
    return (
      <EmptyVault
        handleRefresh={() => {
        }}
        isLoading={false}
      />
    )
  }

  return (
    <ThemedView
      testID='loans_screen'
      style={tailwind('flex-1')}
    >
      <Tabs tabSections={tabsList} testID='loans_tabs' activeTabKey={activeTab} />
      {activeTab === TabKey.YourVaults && <Vaults />}
      {activeTab === TabKey.BrowseLoans && !hasFetchedLoansData &&
      (
        <View style={tailwind('mt-1')}>
          <SkeletonLoader
            row={6}
            screen={SkeletonLoaderScreen.Loan}
          />
        </View>
      )}
      {activeTab === TabKey.BrowseLoans && hasFetchedLoansData &&
      (<LoanCards testID='loans_cards' loans={filteredLoans} />)}
    </ThemedView>
  )
}
