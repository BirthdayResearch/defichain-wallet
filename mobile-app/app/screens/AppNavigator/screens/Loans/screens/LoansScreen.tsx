import * as React from 'react'
import { tailwind } from '@tailwind'
import BigNumber from 'bignumber.js'
import { ThemedView } from '@components/themed'
import { LoanCardOptions, LoanCards } from '@components/LoanCards'

// TODO(pierregee): Remove hardcoded loans once the API is ready
const loans: LoanCardOptions[] = [
  {
  loanName: 'dGOOGL',
  priceType: 'ACTIVE',
  price: new BigNumber('2600.00'),
  isVerified: true,
  interestRate: new BigNumber('2.00'),
  onPress: () => {}
  },
  {
    loanName: 'dTSLA',
    priceType: 'NEXT',
    price: new BigNumber('2500.00'),
    isVerified: false,
    interestRate: new BigNumber('50.00'),
    onPress: () => {}
  }
  ]

export function LoansScreen (): JSX.Element {
  return (
    <ThemedView style={tailwind('h-full')}>
      <LoanCards loans={loans} />
    </ThemedView>
  )
}
