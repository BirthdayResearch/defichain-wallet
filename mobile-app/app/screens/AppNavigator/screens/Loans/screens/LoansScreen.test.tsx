import { render } from '@testing-library/react-native'
import * as React from 'react'
import { LoansScreen } from './LoansScreen'

jest.mock('../../../../../contexts/ThemeProvider')

describe('Loans screen', () => {
  it('should render', async () => {
    const component = (
      <LoansScreen />
    )
    const rendered = render(component)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
