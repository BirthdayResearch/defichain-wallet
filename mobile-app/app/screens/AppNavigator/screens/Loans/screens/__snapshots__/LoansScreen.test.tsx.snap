// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loans screen should render 1`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "flexDirection": "row",
          "paddingTop": 12,
        },
        Object {
          "backgroundColor": "rgba(255, 255, 255, 1)",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "width": "50%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "rgba(255, 0, 175, 1)",
            "borderBottomWidth": 2,
            "borderLeftColor": "rgba(255, 0, 175, 1)",
            "borderRightColor": "rgba(255, 0, 175, 1)",
            "borderTopColor": "rgba(255, 0, 175, 1)",
            "display": "flex",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "RegularFont",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Array [
                  Object {
                    "fontFamily": "SemiBoldFont",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "lineHeight": 24,
                    "paddingBottom": 12,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(0, 0, 0, 1)",
                  },
                ],
              ]
            }
          >
            Browse loans
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "width": "50%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "transparent",
            "borderBottomWidth": 2,
            "borderLeftColor": "transparent",
            "borderRightColor": "transparent",
            "borderTopColor": "transparent",
            "display": "flex",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              undefined,
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "RegularFont",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 24,
                },
                Array [
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                    "paddingBottom": 12,
                    "textAlign": "center",
                  },
                  Object {
                    "color": "rgba(115, 115, 115, 1)",
                  },
                ],
              ]
            }
          >
            Your vaults
          </Text>
        </View>
      </View>
    </View>
  </View>
  <RCTScrollView
    data={
      Array [
        Object {
          "interestRate": "2",
          "isVerified": true,
          "loanName": "dGOOGL",
          "onPress": [Function],
          "price": "2600",
          "priceType": "ACTIVE",
        },
        Object {
          "interestRate": "50",
          "isVerified": false,
          "loanName": "dTSLA",
          "onPress": [Function],
          "price": "2500",
          "priceType": "NEXT",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "marginBottom": -16,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 16,
        },
        Object {
          "backgroundColor": "rgba(245, 245, 245, 1)",
        },
      ]
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderBottomColor": "rgba(229, 229, 229, 1)",
                "borderBottomLeftRadius": 4,
                "borderBottomRightRadius": 4,
                "borderBottomWidth": 1,
                "borderLeftColor": "rgba(229, 229, 229, 1)",
                "borderRightColor": "rgba(229, 229, 229, 1)",
                "borderTopColor": "rgba(229, 229, 229, 1)",
                "borderTopLeftRadius": 4,
                "borderTopRightRadius": 4,
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 16,
                "marginLeft": 8,
                "marginRight": 8,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "paddingBottom": 8,
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={24}
                bbWidth={24}
                focusable={false}
                height={24}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Object {
                      "marginRight": 8,
                    },
                    Object {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={32}
                vbWidth={32}
                width={24}
              >
                <RNSVGGroup>
                  <RNSVGCircle
                    cx={16}
                    cy={16}
                    fill={865906463}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    r={16}
                  />
                  <RNSVGText
                    alignmentBaseline="central"
                    content={null}
                    dx={Array []}
                    dy={Array []}
                    fill={4285168132}
                    font={
                      Object {
                        "fontSize": "24",
                        "fontWeight": "bolder",
                        "textAnchor": "middle",
                      }
                    }
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    rotate={Array []}
                    x={
                      Array [
                        "50%",
                      ]
                    }
                    y={
                      Array [
                        "50%",
                      ]
                    }
                  >
                    <RNSVGTSpan
                      content="D"
                      dx={Array []}
                      dy={Array []}
                      font={Object {}}
                      rotate={Array []}
                      x={Array []}
                      y={Array []}
                    />
                  </RNSVGText>
                </RNSVGGroup>
              </RNSVGSvgView>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontFamily": "SemiBoldFont",
                        "fontWeight": "600",
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      },
                    ],
                  ]
                }
              >
                dGOOGL
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Active Price
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "row",
                  "paddingBottom": 8,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontFamily": "SemiBoldFont",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "lineHeight": 20,
                        "marginRight": 4,
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      },
                    ],
                  ]
                }
              >
                $
                2,600.00
              </Text>
              <Text />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontSize": 12,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "rgba(115, 115, 115, 1)",
                    },
                  ],
                ]
              }
            >
              Interest
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "RegularFont",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "lineHeight": 24,
                  },
                  Array [
                    Object {
                      "fontFamily": "SemiBoldFont",
                      "fontSize": 14,
                      "fontWeight": "600",
                      "lineHeight": 20,
                    },
                    Object {
                      "color": "rgba(0, 0, 0, 1)",
                    },
                  ],
                ]
              }
            >
              2.00%
            </Text>
          </View>
          <View
            style={
              Object {
                "flexBasis": "50%",
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 1)",
                  "borderBottomColor": "rgba(229, 229, 229, 1)",
                  "borderBottomLeftRadius": 4,
                  "borderBottomRightRadius": 4,
                  "borderBottomWidth": 1,
                  "borderLeftColor": "rgba(229, 229, 229, 1)",
                  "borderRightColor": "rgba(229, 229, 229, 1)",
                  "borderTopColor": "rgba(229, 229, 229, 1)",
                  "borderTopLeftRadius": 4,
                  "borderTopRightRadius": 4,
                  "flexBasis": "0%",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginBottom": 16,
                  "marginLeft": 8,
                  "marginRight": 8,
                  "opacity": 1,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "paddingBottom": 8,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  focusable={false}
                  height={24}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Object {
                        "marginRight": 8,
                      },
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={32}
                  vbWidth={32}
                  width={24}
                >
                  <RNSVGGroup>
                    <RNSVGCircle
                      cx={16}
                      cy={16}
                      fill={856658304}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      r={16}
                    />
                    <RNSVGText
                      alignmentBaseline="central"
                      content={null}
                      dx={Array []}
                      dy={Array []}
                      fill={4278415971}
                      font={
                        Object {
                          "fontSize": "24",
                          "fontWeight": "bolder",
                          "textAnchor": "middle",
                        }
                      }
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                      rotate={Array []}
                      x={
                        Array [
                          "50%",
                        ]
                      }
                      y={
                        Array [
                          "50%",
                        ]
                      }
                    >
                      <RNSVGTSpan
                        content="D"
                        dx={Array []}
                        dy={Array []}
                        font={Object {}}
                        rotate={Array []}
                        x={Array []}
                        y={Array []}
                      />
                    </RNSVGText>
                  </RNSVGGroup>
                </RNSVGSvgView>
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Object {
                          "fontFamily": "SemiBoldFont",
                          "fontWeight": "600",
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 1)",
                        },
                      ],
                    ]
                  }
                >
                  dTSLA
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      Object {
                        "color": "rgba(115, 115, 115, 1)",
                      },
                    ],
                  ]
                }
              >
                Next Price
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "flexDirection": "row",
                    "paddingBottom": 8,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "fontFamily": "RegularFont",
                        "fontSize": 16,
                        "fontWeight": "400",
                        "lineHeight": 24,
                      },
                      Array [
                        Object {
                          "fontFamily": "SemiBoldFont",
                          "fontSize": 14,
                          "fontWeight": "600",
                          "lineHeight": 20,
                          "marginRight": 4,
                        },
                        Object {
                          "color": "rgba(0, 0, 0, 1)",
                        },
                      ],
                    ]
                  }
                >
                  $
                  2,500.00
                </Text>
                <Text />
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      Object {
                        "color": "rgba(115, 115, 115, 1)",
                      },
                    ],
                  ]
                }
              >
                Interest
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "fontFamily": "RegularFont",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "lineHeight": 24,
                    },
                    Array [
                      Object {
                        "fontFamily": "SemiBoldFont",
                        "fontSize": 14,
                        "fontWeight": "600",
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 1)",
                      },
                    ],
                  ]
                }
              >
                50.00%
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
